crypto/sessionkey.go

```diff
@@ -99,14 +110,19 @@ func GenerateSessionKeyAlgo(algo string) (sk *SessionKey, err error) {
 }
 
 // GenerateSessionKey generates a random key for the default cipher.
-func GenerateSessionKey() (*SessionKey, error) {
-	return GenerateSessionKeyAlgo(constants.AES256)
+func generateSessionKey(config *packet.Config) (*SessionKey, error) {
+	cf, ok := algosToSymKey[config.DefaultCipher]
+	if !ok {
+		return nil, errors.New("gopenpgp: unsupported cipher function")
+	}
+	return GenerateSessionKeyAlgo(cf)
 }
 
 func NewSessionKeyFromToken(token []byte, algo string) *SessionKey {
 	return &SessionKey{
 		Key:  clone(token),
 		Algo: algo,
+		v6:   algo == "",
```

Comment:
Like this is always false?

Reply:
The v6 flag indicates whether the session key was parsed from a v6 PKESK or SKESK packet. It is needed because v6 PKESK/SKESK packets do not encode the encryption algorithm of the session key. If the flag is set, session key operations, such as encryption, ignore the session key algorithm and determine the algorithm from the configuration.

I assume that if NewSessionKeyFromToken is called, the algorithm is always known?

 I guess if `NewSessionKeyFromToken` is called the algorithm is always known?